### ISSUE TITLE
Fix TypeError with .stylelintrc.js setup

### ISSUE THREAD

**Issue by fossbrandon** [stylelint/stylelint#5714]
### What steps are needed to reproduce the bug?

I am trying to setup stylelint for the first time in a new repository and I am running into an error when I try to lint from the CLI with a `.stylelintrc.js` configuration in the root. My React project is nested further down within the repository. I setup a `package.json` script to perform my linting which is what throws the following error: 

![image](https://user-images.githubusercontent.com/46940941/141217921-62259260-6d71-4f3c-8ff9-87de71f53ec6.png)

**Note:** If I switch the file to `json` format instead of `.js` format, everything works as expected. 

### What Stylelint configuration is needed to reproduce the bug?

`package.json` in a directory other than root:

```json
{
  "scripts": {
    "lint": "stylelint '**/*.css'"
  }
}
```

`.stylelintrc.js` in the root of the repository:

```javascript
module.exports = {
  extends: ["stylelint-config-standard"]
};
```

### How did you run Stylelint?

CLI through `package.json` script with the command `stylelint **/*.css`

### Which version of Stylelint are you using?

`14.0.1`

### What did you expect to happen?

I expect stylelint to use the configuration and perform its linting without throwing a `TypeError`.

### What actually happened?

Running the CLI through a `package.json` script, I get the following error:

```
TypeError: Expected `moduleId` to be of type `string`, got `object`
    at resolveFrom (C:\repo\node_modules\resolve-from\index.js:12:9)
    at Function.module.exports.silent (C:\repo\node_modules\resolve-from\index.js:47:54)
    at getModulePath (C:\repo\node_modules\stylelint\lib\utils\getModulePath.js:16:25)
    at C:\repo\node_modules\stylelint\lib\augmentConfig.js:129:60
    at Array.map (<anonymous>)
    at absolutizePaths (C:\repo\node_modules\stylelint\lib\augmentConfig.js:129:44)
    at augmentConfigBasic (C:\repo\node_modules\stylelint\lib\augmentConfig.js:47:9)
    at async augmentConfigFull (C:\repo\node_modules\stylelint\lib\augmentConfig.js:90:24)
    at async run (C:\repo\node_modules\cosmiconfig\dist\Explorer.js:42:31)
    at async cacheWrapper (C:\repo\node_modules\cosmiconfig\dist\cacheWrapper.js:16:18)
```

### Does the bug relate to non-standard syntax?

No

**Comment by ybiquitous** [comment]
@fossbrandon Thank you for using a template!

I try the following reproduction, but it never raises an error. 🤔 
Could you let us your environment and provide a small reproduction repository, please?

```console
$ cat a.css
.foo {
}

$ cat .stylelintrc.js
module.exports = {
  extends: ["stylelint-config-standard"]
};

$ cat package.json
{
  "dependencies": {
    "stylelint": "^14.0.1",
    "stylelint-config-standard": "^23.0.0"
  },
  "scripts": {
    "lint": "stylelint '**/*.css'"
  }
}

$ npm run lint

> lint
> stylelint '**/*.css'


a.css
 1:6  ✖  Unexpected empty block  block-no-empty

$ node -v
v16.13.0

$ npm -v
8.1.3
```

**Comment by fossbrandon** [comment]
Thanks for taking a look! I'll go ahead and see if I can reproduce this in a new repo and get back to you.

**Comment by fossbrandon** [comment]
While I was creating a new repo to reproduce the issue I figured out what was wrong. It ended up being a silly mistake on my end 😅 

My `.stylelintrc.js` file originally had some empty placeholder configuration options such as `rules` and `plugins` but I specified `plugins` with curly braces instead of brackets...

Original which broke:

```javascript
module.exports = {
  extends: ["stylelint-config-standard"],
  rules: {},
  plugins: {},
};
```

Corrected which works:

```javascript
module.exports = {
  extends: ["stylelint-config-standard"],
  rules: {},
  plugins: [],
};
```

Thank you @ybiquitous for helping me look into it! I'll go ahead and close this issue.
